packages/eslint-plugin/docs/rules/no-unsafe-unary-minus.md

```diff
@@ -0,0 +1,37 @@
+---
+description: 'Require unary negation to take a number.'
+---
+
+> 🛑 This file is source code, not the primary documentation location! 🛑
+>
+> See **https://typescript-eslint.io/rules/no-unsafe-unary-minus** for documentation.
+
+TypeScript does not prevent you from putting a minus sign before things other than numbers:
+
+```ts
+const s = 'hello';
+const x = -s; // x is NaN
+```
+
+This rule restricts the unary `-` operator to `number | bigint`.
+
+## Examples
```

Comment:
[Docs] I like that this shows a more real-world usage (functions) than just literals. But it's a bit more to read when they're all functions. What do you think about making most of the examples literals?

Another syntax I like to use in cases like this is `declare`:

```ts
declare const eitherNumeric: number | bigint;
-eitherNumeric;
```

Reply:
OK! I replaced all the functions with `declare`s.